consumer_group.go

```diff
@@ -331,20 +331,14 @@ func (c *consumerGroup) syncGroupRequest(coordinator *Broker, plan BalanceStrate
 		MemberId:     c.memberID,
 		GenerationId: generationID,
 	}
+	strategy := c.config.Consumer.Group.Rebalance.Strategy
 	for memberID, topics := range plan {
 		assignment := &ConsumerGroupMemberAssignment{Topics: topics}
-
-		// Include topic assignments in group-assignment userdata for each consumer-group member
-		if c.config.Consumer.Group.Rebalance.Strategy.Name() == StickyBalanceStrategyName {
-			userDataBytes, err := encode(&StickyAssignorUserDataV1{
-				Topics:     topics,
-				Generation: generationID,
-			}, nil)
-			if err != nil {
-				return nil, err
-			}
-			assignment.UserData = userDataBytes
+		userDataBytes, err := strategy.AssignmentData(plan, memberID, generationID)
```

Comment:
wondering why you'd need to send the plan and member to `AssignmentData` to get the topics:
``` go
	topics, ok := plan[memberID]
	if !ok {
		return nil, nil
	}
```
if that was already obtained in the range above?
Line 335 `for memberID, topics := range plan`

Reply:
My thinking here was in the case the state being shipped needed to understand data about the larger plan, but I am happy limiting the scope to just the given member and its assigned topics.

Reply:
thanks for the fast response!
I'll 👀 later today